agent_app_sample_code/cookbook_utils/cookbook_config.py

```diff
@@ -0,0 +1,203 @@
+              
+              self.mlflow_experiment_name = f"{user_home_directory}/{self.uc_asset_prefix}_mlflow_experiment"
+            except Exception as e:
+            #   print(e)
+              raise ValueError(f"Failed to identify the current user's working directory, which is used to initialize the default value for `mlflow_experiment_name`.  Please explicitly specify `mlflow_experiment_name` and retry.")
+
+    def are_any_uc_asset_names_empty(self) -> bool:
+        """
+        Check if any of the Unity Catalog asset names are empty.
+
+        Returns:
+            bool: True if any of the asset names (`mlflow_experiment_name`, `uc_model`, or `evaluation_set_table`) are None, otherwise False.
+        """
+        if (
+            self.mlflow_experiment_name is None
+            or self.uc_model is None
+            or self.evaluation_set_table is None
+        ):
+            return True
+        else:
+            return False
+          
+    def get_uc_fqn(self, asset_name: str) -> str:
+        """
+        Generate the fully qualified name (FQN) for a Unity Catalog asset.
+
+        Args:
+            asset_name (str): The name of the asset to generate the FQN for.
+
+        Returns:
+            str: The fully qualified name of the asset, with necessary escaping for special characters.
+        """
+        uc_fqn = f"{self.uc_catalog_name}.{self.uc_schema_name}.{self.uc_asset_prefix}_{asset_name}"
+
+        return self.escape_uc_fqn(uc_fqn)
+
+    def get_uc_fqn_for_asset_name(self, asset_name):
+        uc_fqn = f"{self.uc_catalog_name}.{self.uc_schema_name}.{asset_name}"
+
+        return self.escape_uc_fqn(uc_fqn)
+
+    def escape_uc_fqn(self, uc_fqn):
+        """
+        Escape the fully qualified name (FQN) for a Unity Catalog asset if it contains special characters.
+
+        Args:
+            uc_fqn (str): The fully qualified name of the asset.
+
+        Returns:
+            str: The escaped fully qualified name if it contains special characters, otherwise the original FQN.
+        """
+        if "-" in uc_fqn:
+            parts = uc_fqn.split(".")
+            escaped_parts = [f"`{part}`" for part in parts]
+            return ".".join(escaped_parts)
+        else:
+            return uc_fqn
+    
+
+    def dump_to_yaml(self, file_path: str):
+        import yaml
+        with open(file_path, 'w') as file:
+            yaml.dump(self.model_dump(), file)
+
+    @classmethod
+    def from_yaml_file(self, file_path: str):
+        import yaml
+        with open(file_path, 'r') as file:
+            data = yaml.safe_load(file)
+            # print(data)
+        config = AgentCookbookConfig.parse_obj(data)
+        # print(config.model_dump())
+        return config
+    
+    def pretty_print(self):
+        config_dump = self.model_dump()
+        print(json.dumps(config_dump, indent=4))
+      
+
+    def validate_or_create_uc_catalog(self) -> bool:
```

Comment:
I found the default error message from the SDK in case of permission failures is actually decent, e.g.:

```
PermissionDenied: User does not have CREATE SCHEMA and USE CATALOG on Catalog 'default'. Config: host=https://oregon.staging.cloud.databricks.com/, auth_type=runtime
```

Will propose a simplification to this code (I think the print statements when we're attempting to create are still useful, but we probably don't need to wrap the permission denied case) that reduces the # of branches to test/maintain etc